Taskfile.yaml

```diff
@@ -0,0 +1,224 @@
+  PRIVATE_KEY_DEFAULT: '{{.ARTIFACT_DIR}}/testkey'
+  PUBLIC_KEY: '{{ .PUBLIC_KEY | default .PUBLIC_KEY_DEFAULT}}'
+  PRIVATE_KEY: '{{ .PRIVATE_KEY | default .PRIVATE_KEY_DEFAULT}}'
+  DEPLOYMENT_ID: '{{default "devex-cicd" .DEPLOYMENT_ID}}'
+  PATH="{{.ANSIBLE_INSTALL_DIR}}/bin/:{{.TERRAFORM_INSTALL_DIR}}/bin/:$PATH":
+  NUM_NODES: '{{default "3" .NUM_NODES}}'
+
+  TF_SSH_USER_VAR:
+    sh: |
+      # ssh_user variable is expected for gcp
+      if [[ "{{.CLOUD_PROVIDER}}" == "gcp" ]]; then
+        echo "-var='ssh_user={{default .USER .TF_SSH_USER}}'"
+      fi
+
+env:
+  TF_IN_AUTOMATION: '{{.CI}}'
+  TF_DATA_DIR: '{{.CLOUD_PROVIDER}}'
+
+tasks:
+  keygen:
+    desc: generate ssh keys
+    summary:
+      Generates some local dummy keys to support running with terraform and ansible in CI and to try things out. 
+      If you have your own keys you should use them instead.
+    run: always
+    vars:
+      SSH_EMAIL: '{{default "test@redpanda.com" .SSH_EMAIL | lower}}'
+    cmds:
+    - | 
+      if [[ ! -f artifacts/testkey ]]; then
+          ssh-keygen -t rsa -b 4096 -C "{{.SSH_EMAIL}}" -N "" -f artifacts/testkey <<< y
+      fi
+
+  install-prereqs:
+    desc: install ansible and terraform
+    summary: |
+      Convenience feature that installs terraform and ansible to the artifact directory. 
+      For normal operation you will also likely need curl and unzip.
+      This is not a part of the flow, it must be called explicitly.
+    vars:
+      TF_URL: https://releases.hashicorp.com/terraform/{{.TERRAFORM_VERSION}}/terraform_{{.TERRAFORM_VERSION}}_{{OS}}_{{ARCH}}.zip
+    cmds:
+      - mkdir -p '{{.TERRAFORM_INSTALL_DIR}}/bin'
+      - rm -f '{{.TERRAFORM_INSTALL_DIR}}/bin/terraform'
+      - curl -L '{{.TF_URL}}' -o '{{.TERRAFORM_INSTALL_DIR}}/bin/tf.zip'
+      - unzip '{{.TERRAFORM_INSTALL_DIR}}/bin/tf.zip' -d '{{.TERRAFORM_INSTALL_DIR}}/bin'
+      - rm '{{.TERRAFORM_INSTALL_DIR}}/bin/tf.zip'
+      - python3 -mvenv '{{.ANSIBLE_INSTALL_DIR}}'
+      - |
+        source '{{.ANSIBLE_INSTALL_DIR}}/bin/activate'
+        pip3 install --upgrade pip
+        pip3 install ansible-core=={{.ANSIBLE_VERSION}}
+    status:
+      - test -f '{{.TERRAFORM_INSTALL_DIR}}/bin/terraform || command -v terraform'
+      - test -f '{{.ANSIBLE_INSTALL_DIR}}/bin/ansible || command -v ansible'
+
+  apply:
+    desc: init and apply terraform
+    summary: |
+      You can set additional arguments to be consumed by terraform for building out the cluster by adding them after 
+      '--'when calling the task. You can set the number of nodes by setting NUM_NODES
+      You may also want to set a backend for your runs if you are using something like Terraform Cloud to manage state. 
+      See the commented out line in cmds for an example of how to achieve this
+    env:
+     AWS_ACCESS_KEY_ID: '{{.DA_AWS_ACCESS_KEY_ID}}'
+     AWS_SECRET_ACCESS_KEY: '{{.DA_AWS_SECRET_ACCESS_KEY}}'
+     AWS_DEFAULT_REGION: '{{ default "us-west-2" .AWS_DEFAULT_REGION}}'
```

Comment:
note that, due to the way go-task works, environment variables that exist prior to the execution of task cannot be overridden (see https://github.com/go-task/task/discussions/449). when this executes locally, if `AWS_*` exist, they won't be overridden by the values of `DA_*`. it might be worth adding a note or make it part of the `summary:`

Reply:
ok. that's the intended way things should work so I'll note it 👍